src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +696,99 @@ export class ResidualHeapSerializer {
     return t.arrayExpression(initProperties);
   }
 
+  _getPropertyValue(object: any, key: string, serialzeValue: boolean) {
+    if (object != null && object.has(key) === true) {
+      let val = object.get(key);
+
+      if (val !== undefined) {
+        let descriptor = val.descriptor;
+
+        if (descriptor !== undefined) {
+          let descriptorValue = descriptor.value;
+
+          if (descriptorValue !== undefined) {
+            return serialzeValue === true ? this.serializeValue(descriptorValue) : descriptorValue;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  _serializeValueReactElement(val: ObjectValue): BabelNodeExpression {
+    let objectProps = val.properties;
+
+    // we do this so they don't get emited
+    this.serializedValues.add(val.$Prototype);
```

Comment:
Don't mutate `serializedValues` like this. Instead, change the visitor to not visit the value.

Reply:
A few reasons:
- By pulling on this value, the visitor will transitively visit everything that this value depends on. Just adding the value to `serializedValues` doesn't do the transitive serialization, leading to obscure invariant failures eventually.
- Visiting values in scopes where they are not needed will lead to suboptimal code generation.